### ISSUE TITLE
MySQL Socket access in services other than database

### ISSUE THREAD

**Issue by com2** [lando/lando#2262]
Reading the MySQL documentation about overriding its configuration, I discovered that the default configuration file can be found inside the service database at /opt/bitnami/mysql/conf/my.cnf So I did "lando -s database ssh" and then "cat /opt/bitnami/mysql/conf/my.cnf" and found out that the socket path is /opt/bitnami/mysql/tmp/mysql.sock . Rather naievely I then thought that I could use that path in the database configuration of Drupal.  Drupal warns "The website encountered an unexpected error. Please try again later." and then I checked that in appserver the path does not exist. 

What do I need to change in .lando.yml to make the mysql socket path available in appserver for use in Drupal's database configuration?

**Comment by pirog** [comment]
@com2 you could volume mount the socket and share it between containers. However it would be much easier to either:

Run `lando info` and use the the internal connection/creds information
or
Read the section in the Drupal 8 documentation that is called "Connecting to your database"
https://docs.lando.dev/config/drupal8.html#connecting-to-your-database
or
Look at the Drupal 8 example code
https://github.com/lando/lando/tree/master/examples/drupal8

**Comment by com2** [comment]
Well, yes I know how to connect to a database both through tcp as through sockets. That is not the problem. 

I want to use socket database connection in the production site and lando info does not provide me the information. Some contributed modules appeared to be incompatible with socket and therefore I would like to use it in my Lando stage too. 

I tried the following:
```yaml
name: fotod8
recipe: drupal8
config:
  webroot: web
services:
  database:
    type: mariadb
    config:
      database: .lando/my-custom.cnf
```
.lando/my-custom.cnf:
```
[mysqld]
socket=/app/.lando/tmp/mysql.sock
```
The database will start after rebuilding, but in the database startup log I don't see inclusion of my custom configuration and it ends the start log with: 
```
database_1   | Version: '5.7.29'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL)
```
...which means the requested socket change was not included. 

How should I proceed?
